compiler/rustc_mir_build/src/build/matches/util.rs

```diff
@@ -24,6 +24,51 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
             .collect()
     }
 
+    #[instrument(skip(self), level = "debug")]
+    pub(crate) fn field_match_pairs_tuple_struct<'pat>(
```

Comment:
why is this only for tuple structs, doesn't this issue also affect other field pattern?

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=836049e2bb4d4b9b61ba4c32e0bf4a3f

```rust
struct Foo<T> {
    x: T,
} // `T` is covariant.

fn foo<'b>(x: Foo<for<'a> fn(&'a ())>) {
    let Foo { x }: Foo<fn(&'b ())> = x;
}

fn foo_nested<'b>(x: Foo<Foo<for<'a> fn(&'a ())>>) {
    let Foo { x: Foo { x } }: Foo<Foo<fn(&'b ())>> = x;
}

fn main() {}
```